libs/application/templates/family-matters/joint-custody-agreement/types/index.ts

```diff
@@ -0,0 +1,31 @@
+import { Application, FieldBaseProps } from '@island.is/application/core'
+import {
+  MockData,
+  ExternalData,
+  Override,
+  NestedType,
+} from '@island.is/application/templates/family-matters/types'
+import { answersSchema } from '../lib/dataSchema'
+
+// We are using mockData that is not defined in the zod schema
+export interface Answers extends answersSchema {
+  mockData: MockData
+}
+
+export type JCAApplication = Override<
+  Application,
+  { answers: Answers; externalData: ExternalData }
+>
+
+type ErrorSchema = NestedType<answersSchema>
+
+export type JCAFieldBaseProps = Override<
+  FieldBaseProps,
+  { application: JCAApplication; errors: ErrorSchema }
+>
+
+export enum DataProviderTypes {
+  MockNationalRegistry = 'MockNationalRegistryProvider',
+  NationalRegistry = 'NationalRegistryProvider',
+  UserProfile = 'UserProfileProvider',
+}
```

Comment:
Talked about this in IRL but maybe this type will be moved to shared later and the providers as well.

Reply:
Yes, just want this PR merged to get code ownership and then we can move things around